.travis.yml

```diff
@@ -0,0 +1,15 @@
+language: python
+cache:
+  pip: true
+  directories:
+  # - datasets
+before_install:
+  - python --version
+  - pip install -U pip
+  - pip install codecov
+install:
+    - pip install -e .[test] # install package + test dependencies
+script:
+    - pytest --cov=time_series_transformer # run tests
+after_success:
+  - codecov # submit coverage 
```

Comment:
I would rather avoid using Travis here, this is not so much a development project that it requires CI/CD.

Reply:
It is a good practice to whenever there is a code change, to run a checklist of all the desired functionality covering all corners of the code base. If you look at [codecov](https://codecov.io/gh/DanielAtKrypton/time_series_transformer) with the [preliminary tests](https://github.com/DanielAtKrypton/time_series_transformer/blob/master/tests/main_test.py) I came up with 65% of the code was already covered. Now that number can reach 100% with the addition of a few more tests, so that you can see it is always working as intended when there is a code change. And if the build breaks for any reason you are immediatelly informed by email. Then you can go there and fix it fast. Sometimes a dependency breaks the code base, as was the case with [numpy 1.19.4](https://stackoverflow.com/questions/64654805/how-do-you-fix-runtimeerror-package-fails-to-pass-a-sanity-check-for-numpy-an) in Windows. The only solution as of now was to stick with a previous numpy version.

Reply:
Yes I understand, but we currently don't have any kind of test, and dependency problems are left to the user by not mentioning the precise version of each package in the requirements.

If at some point we develop some meaningful tests, we could consider using Travis, but wright now it's just a waste of CPU time.

Reply:
I developed two small fast tests for it. One with the GPU if it is available and another with the CPU. See [here](https://github.com/DanielAtKrypton/time_series_transformer/blob/master/tests/main_test.py#L29). There is a notebook that shows the test cenario [here](https://time-series-predictor.readthedocs.io/en/latest/notebooks/example_flights_dataset.html). This file was also removed in the latest pull request draft.